index.js

```diff
@@ -40,33 +71,72 @@ function Blockchain (opts) {
   })
 }
 
+/**
+ * Define meta getter for backwards compatibilty
+ */
+Blockchain.prototype = {
+  get meta () {
+    return {
+      rawHead: this._headHeader,
+      heads: this._heads,
+      genesis: this._genesis
```

Comment:
The ``meta`` ``getter`` is not really returning the meta like in the previous version and as expected after reading your point 3) description point:

> As a result, the meta field has been moved into a getter that generates the old meta info from other internal fields

Old format:

```javascript
{ heads: {},
  td: <BN: 400000000>,
  rawHead: 'd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3',
  height: 0,
  genesis: 'd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3' }
```

New format:
```javascript
{ rawHead: <Buffer d4 e5 67 40 f8 76 ae f8 c0 10 b8 6a 40 d5 f5 67 45 a1 18 d0 90 6a 34 e6 9a ec 8c 0d b1 cb 8f a3>,
  heads: {},
  genesis: <Buffer d4 e5 67 40 f8 76 ae f8 c0 10 b8 6a 40 d5 f5 67 45 a1 18 d0 90 6a 34 e6 9a ec 8c 0d b1 cb 8f a3> }
```

So types are different and ``td`` and ``height`` are missing. Is this intentional?

Reply:
> while this is supporting the old constructors, it won't be possible with this to use an already written DB any more (this is correct, isn't it?)

Yes, this PR is not compatible with existing DBs and we should definitely make that very clear. I could create a script to migrate old DBs into geth format if there is enough demand for it.

> So types are different and td and height are missing. Is this intentional?

The td and height were intentionally left out. The README doesn't mention a meta field and it refers to BlockChain Properties that don't exist so I was unsure whether removing meta would break the published interface. My preference would be to remove the meta field completely and explicitly expose certain properties (headHeader, headBlock, genesis) and async get methods such as getTd(cb) and getHeight(cb). The async methods are needed since computing td and height require db operations under the geth db design. However, if we absolutely needed to keep the current meta interface and make these values available synchronously, it is possible but would require additional db calls to ensure these values are always up-to-date. My opinion is that the convenience of this doesn't outweigh the additional performance overhead of preemptively computing these values whenever there is a change to the blockchain (instead of computing them on-demand). If we don't want to break existing dependencies, I could fix the meta getters to return correct values (including pre-computed td and height), but also deprecate meta and add new properties and async get methods to the interface going forward.

Regarding the difference in types for meta.rawHead and meta.genesis, that was a mistake on my part! Internally, I keep all hash values as Buffers until a conversion to String was absolutely necessary. Its really easy to change the getter to return hex strings instead.

Reply:
1) DB compatibility
I would assume that it won't become necessary but it's good to know that there is this fallback solution with a migration script in case there are more people relying on this then realized. People can also still use the ``v2.1.0`` version (for some time).

2)meta 
I would also say that we can drop the ``meta`` "interface" completely. This was always something very implicit, just did a short GitHub search, within the ``ethereumjs`` ecosystem I found only two direct accesses on this from within the VM implementation which can be easily updated. I very much prefer your solution to expose these properties directly in the way you described above.